docs/presentations/bmo-deep-dive/bmo-deep-dive.html

```diff
@@ -0,0 +1,409 @@
+
+
+
+# Introduction
+
+
+
+The Bare Metal Operator or BMO implements a Kubernetes API for managing bare metal hosts. It maintains an inventory of available hosts as instances of the BareMetalHost Custom Resource Definition. The Bare Metal Operator knows how to:
+* Inspect the host’s hardware details such as CPUs, RAM, disks, NICs, etc. and report them on the corresponding BareMetalHost
+* Provision hosts with a desired image and clean a host’s disk contents before or after provisioning
+
+<br>
+
+Bare Metal Operator abstracts hardware ( at present physical nodes ) and provides them as consumables to the Kubernetes [Cluster API](https://github.com/kubernetes-sigs/cluster-api) via **Metal3’s** [Cluster API Provider Metal3](https://github.com/metal3-io/cluster-api-provider-metal3) which extends the Cluster API as the provider which handles real hardware.
+<br>
+<br>
+See this [BMO state diagram](https://github.com/metal3-io/baremetal-operator/blob/master/docs/baremetalhost-states.md) describing various states part of BMO’s lifecycle, the different states are briefly described below the diagram in this document.
+<br>
+<br>
+BMO is integral to the Metal3 project and provides a very useful functionality of using real hardware for Kubernetes deployment. But it is also useful for standalone use cases and we will discuss that later in the presentation.
+Below is a diagram depicting BMO part of a Metal3 environment
+
+
+
+### Representation of BMO in Metal3 environment
+
+<img data-src="bmo-in-metal3.png" height="60%" width="70%" />
+
+
+
+
+### BMO with Metal3
+* The **baremetal-operator** is the component that manages bare metal hosts. It exposes a new BareMetalHost custom resource in the Kubernetes API that lets you manage hosts in a declarative way.
+<br />
+<br />
+<br />
+<br />
+<br />
+* **cluster-api-provider-metal3** is the component that provides integration with the cluster-api project. This provider includes a Metal3Machine object that acts as a client of the BareMetalHost custom resources (CR).
+
+
+
+# API
+
+
+
+The BareMetalHost embeds two well differentiated sections, the bare metal host specification and its current status.
+<br>
+<br>
+These two sections being :
+* BareMetalHost spec
+* BareMetalHost status
+
+There are many fields under both these sections, we shall look at few of the key fields among them.
+<br>
+<br>
+First let’s look at a sample BareMetalHost CRD ( Custom Resource Definition ), defining a simple host definition and containing a part of all the available fields.
+<br>
+<br>
+Then we will take a look at sections of a BareMetalHost Object after its creation like Spec, Status, etc.
+<br>
+<br>
+For further details about the BMO API, refer the [documentation here](https://github.com/metal3-io/baremetal-operator/blob/master/docs/api.md).
+
+
+
+### Sample BareMetalHost Definition
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bmo-master-0-bmc-secret
```

Comment:
This does not match with the secret name provided in BMH later, it should be 
```suggestion
  name: bm0-bmc-secret
```